packages/core/src/test/shared/utilities/pathFind.test.ts

```diff
@@ -42,4 +46,46 @@ describe('pathFind', function () {
         const regex = /bin[\\\/](code|code-insiders)$/
         assert.ok(regex.test(vscPath), `expected regex ${regex} to match: "${vscPath}"`)
     })
+
+    describe('findSshPath', function () {
+        let tryRunStub: sinon.SinonStub
+        // On Windows the first call to tryRun gives a valid path, which changes func behavior.
+        // We can offset our call checks based on this.
+        let callOffset: number
+
+        before(function () {
+            tryRunStub = sinon.stub(programUtils, 'tryRun')
+            callOffset = isWin() ? 1 : 0
+        })
+
+        after(function () {
+            tryRunStub.restore()
+        })
+
+        it('first tries $PATH', async function () {
+            tryRunStub.onCall(callOffset).resolves(true)
```

Comment:
a way to test this without stubs would be to set $PATH to a test fixture directory.

Reply:
I see. I was nervous about having code messing with $PATH in case something went wrong. Tried to safely wrap it in a reusable util function and added some tests for it. I think with this utility, the tests feel much more useful, so thanks for pushing me in this direction.